include/scf/driver/commutator.hpp

```diff
@@ -0,0 +1,4 @@
+#pragma once
+#include <simde/simde.hpp>
+
+simde::type::tensor commutator(simde::type::tensor Fock_Matrix, simde::type::tensor Density_Matrix, simde::type::tensor Overlap_Matrix);
```

Comment:
```suggestion
simde::type::tensor commutator(simde::type::tensor A, simde::type::tensor B, simde::type::tensor C);
```

This is a general function, so let's use generic matrix labels. I can go either way on the capitalization of the argument. Our standards say they should be lowercase, but matrices are traditionally capitalized (so arguably capitalizing them improves readability).

Reply:
Should I make the function name more specific then? I know we're going to make this into a module at some point but I think this function as currently implemented is still highly specific to the scf procedure due to needing an overlap matrix. I mean I could always put a check in there that if C is not defined, assume that A and B are orthogonalized (I think).

Reply:
Why do you think this function is highly specific?

Reply:
Well, I'm mostly meaning that the inputs are specific I guess. The function always assumes that the final matrix is the overlap matrix, plus I pulled the code right out of the scf loop. Should I rewrite this as a generalized commutation function that evaluates each matrix to see which one might be the overlap, if supplied? Otherwise just supply an identity matrix that has the same size as the input matrices?

Reply:
The measure of interaction between two basis vectors. By overlap in the third sentence, I mean which one might be the overlap matrix. I think I'm barking up the wrong tree though.

Reply:
You're fine I misinterpreted your previous statement. Yeah I think assuming the last argument is the overlap matrix is fine. 

This function is general. For matrices A and B and a metric (generalization of the overlap matrix) C, the commutator is defined as ABC-CBA. If the basis set is orthonormal, C=1, and you recover AB-BA.